otcextensions/sdk/cloud/cce.py

```diff
@@ -416,3 +420,381 @@ def delete_cce_cluster_node(
+                'value': 'value2
+            }]
+        :param list taints: List of taints
+            [{
+                'key': 'key1',
+                'value': 'value1,
+                'effect': 'NoSchedule
+            },{
+                'key': 'key2',
+                'value': 'value2,
+                'effect': 'NoSchedule
+            }]
+            Effect available options: NoSchedule,
+            PreferNoSchedule, NoExecute.
+
+        :returns: The results of CCE node pool creation
+        :rtype: :class:`~otcextensions.sdk.cce.v3.node_pool.NodePool`
+        """
+        count = kwargs.get('count')
+        cce_cluster = kwargs.get('cluster')
+        ecs_group = kwargs.get('ecs_group')
+        flavor = kwargs.get('flavor')
+        k8s_tags = kwargs.get('k8s_tags')
+        lvm_config = kwargs.get('lvm_override_config')
+        min_node_count = kwargs.get('min_node_count')
+        max_node_count = kwargs.get('max_node_count')
+        max_pods = kwargs.get('max_pods')
+        name = kwargs.get('name')
+        node_image_id = kwargs.get('node_image_id')
+        os = kwargs.get('os')
+        preinstall_script = kwargs.get('preinstall_script')
+        postinstall_script = kwargs.get('postinstall_script')
+        priority = kwargs.get('priority')
+        public_key = kwargs.get('public_key')
+        scale_down_cooldown_time = kwargs.get('scale_down_cooldown_time')
+        ssh_key = kwargs.get('ssh_key')
+        network_id = kwargs.get('network_id')
+        tags = kwargs.get('tags')
+        taints = kwargs.get('taints')
+
+        volume_types = ['SAS', 'SATA', 'SSD']
+        effects = ['NoSchedule', 'PrefereNoSchedule', 'NoExecute']
+
+        metadata = {
+            'name': name,
+        }
+
+        # Node template specs
+        node_template = {
+            'dataVolumes': {},
+            'extendParam': {},
+            'k8sTags': {},
+            'login': {},
+            'rootVolume': {},
+            'userTags': [],
+            'taints': [],
+            'nodeNicSpec': {
+                'primaryNic': {}
+            }
+        }
+
+        if not (cce_cluster
+                and flavor
+                and os
+                and name
+                and network_id
+                and ssh_key):
+            raise ValueError('One or more of the following required '
+                             'arguments are missing: cce_cluster, '
+                             'flavor, name, network_id, os, ssh_key')
+        node_template['flavor'] = flavor
+        node_template['az'] = availability_zone
+        if count:
+            if not isinstance(count, int):
+                raise ValueError('count is not an integer value')
+            if count < 0:
+                raise ValueError('count is 0 or lower')
+            node_template['count'] = count
+        node_template['login']['sshKey'] = ssh_key
+        node_template['os'] = os
+        if billing_mode and (billing_mode != 0):
```

Comment:
should parentheses be here? `(billing_mode != 0):`

Reply:
billing_mode is an integer value. So parentheses would expect a string... I think or did I missunderstood you?

Reply:
oh, understood